common.mk

```diff
@@ -40,11 +65,34 @@ ER_BPF_BUILDDIR := lib/bpf/bytecode
 CLANG_BPF_SYS_INCLUDES = $(shell $(CLANG) -v -E - </dev/null 2>&1 \
 	| sed -n '/<...> search starts here:/,/End of search list./{ s| \(/.*\)|-idirafter \1|p }')
 
-STATIC_LIBS += -L/usr/libbpf-${LIBBPF_VER}/lib64/ -lbpf -lelf -lz
-# Link static version of libraries required by Teleport (bpf, pcsc) to reduce system dependencies. Avoid dependencies on dynamic libraries if we already link the static version using --as-needed.
-CGOFLAG = CGO_ENABLED=1 CGO_CFLAGS="-I/usr/libbpf-${LIBBPF_VER}/include" CGO_LDFLAGS="-Wl,-Bstatic $(STATIC_LIBS) -Wl,-Bdynamic -Wl,--as-needed"
+# Add library path for CentOS 7 buildbox.
+# TODO(camh): Remove when that buildbox is decomissioned.
+STATIC_LIBS += -L/usr/libbpf-${LIBBPF_VER}/lib64/
+
+# gcc does not use LIBRARY_PATH when cross-compiling, so explicitly add those
+# paths to the linker command line.
+STATIC_LIBS += $(addprefix -L,$(subst :, ,$(LIBRARY_PATH)))
+
+# Libraries needed for or by BPF
+STATIC_LIBS += -lbpf -lelf -lz
```

Comment:
FYI, those are not the correct libraries all the time. libelf added in newer versions dependency on libzstd, so I used pkg-conf to get the "correct" version. Otherwise I'm using the same set to not break build on systems without pkg-conf.

As I mentioned a few times IMO we should merge https://github.com/gravitational/teleport/pull/36106 and stop dealing with libbpf and all related issues.
CC @r0mant

Reply:
I've added a libelf.pc file to buildbox-ng so I should be able to use your refactoring here and not hard-code these libraries unless we do not find a libelf.pc at all.